app/models/notification.rb

```diff
@@ -35,6 +33,11 @@ def self.logger(extra_fields = {})
 
   scope :due_today, -> { where(remind_on: Date.current, status: [:pending]) }
 
+  def record_communication(communication)
```

Comment:
It seems a bit strange that this responsibility lives on the notification model. The channel creates the detail and communication, right? Can the dispatch service orchestrate this "linkage"?

Reply:
This was moved here since:

Notification is already aware of communications (and that it can have many).
Attaching a communication and changing the status to sent is an atomic operation (not in the transactional meaning). We don't want a notification to have a communication but not be marked sent. This is something the model should handle and is too much responsibility for the service.